### DISCUSSION TITLE
Modern UI

### DISCUSSION THREAD

**Discussion by Lenart12** [XCSoar/XCSoar#1050] [Ideas]
Hello,
I created a design prototype of how a modern design could look.

![](https://i.imgur.com/nwmj0kP.png)

This would only change the menu part of the app, nothing with map and info boxes since that already looks good :^)

Before I would make more of the mockup design I am looking for feedback if this is the direction that this app wants to go.

The icon pack is [Material ](https://mui.com/material-ui/material-icons/) which is under Apache v2.0 license so no problem including it in the project.

**Comment by MaxKellermann** [comment]
How will this look on very low-res/small screens and how in landscape mode?

**Comment by lordfolken** [comment]
Constraints to consider;
 * We support screens as small as 320x200
 * We support landscape/portrait
 * A large amount of users use Kobo E-readers (E-ink, so avoid scrolling)
 * Keep it keyboard/joystick operateable (up/down/left/right,ok,cancel)
 * Keep it glove operatable. So no tiny selection sizes.
 * High contrast preferred
 * Minimize head down time (current uis are designed to be attention grabbers)

What i like about that proposal:
 - 3 Font Sizes 
 - iconography (we should use that more.)
 - Personally i like the sleek look of number 3. 
 
What i don't like:
 - The setting values are invisible. This is imo worse than what we currently have.
 - Everything on one screen will be difficult with the constraints we are working with (see above)


Suggestions:
 * Do it like the android settings:
 Top menu with all the sections -> Fold out  baesd on selection, -> One page per section -> Drop to settings (if needed)
![image](https://user-images.githubusercontent.com/407371/190827141-2bbc7b14-8241-4449-aacd-4ecf426328f4.png)

**Reply by hjr** [reply]
Happy to see this discussion here. A visual boost to XCS is due and would demonstrate the project alive, even it is a minor code change ..
my 2cts here

- The one big and very first argument I hear when people report having dropped CU from their toys in favor of XCS is is the better touch screen operability of XCS with larger buttons and widgets. The current config menu does follow this touch operation principle very well. It is an important requirement I see not changing in the future for this app.

- The structured settings do organize them in a max. 7 items per screen or selector, which is a well known design rule of thumb of what our brains are willing to capture at a glance and at once. Also some thing we should have in mind structuring, organizing these things, have been well done for XCS currently.

- I do like the structure of your config menu screenshot. Top level items could hide there second level item list until you select them. This would reduce the over all number of items on the first selector screen and enlarge the tourch area. Once selected show just the second level item list while collapsing all other to level items. You then see your navigation confirmed finding you desired second level item. Then another touch interaction on the item list incl. the top level item actually opens the sub menu and presents the enlarged item list for further selection. A third touch enters then your config item. Three touch interaction instead of two, enable the larger touch areas.
